vach/src/loader/archive.rs

```diff
@@ -201,20 +180,32 @@ where
 
 		Ok(Archive {
 			header,
-			handle,
+			handle: Arc::new(Mutex::new(handle)),
 			key: config.public_key,
 			entries,
 			decryptor,
 		})
 	}
 
 	#[inline(always)]
-	pub(crate) fn fetch_raw(&mut self, entry: &RegistryEntry) -> InternalResult<Vec<u8>> {
-		let handle = &mut self.handle;
-		handle.seek(SeekFrom::Start(entry.location))?;
+	pub(crate) fn fetch_raw(&self, entry: &RegistryEntry) -> InternalResult<Vec<u8>> {
+		let handle = &self.handle;
+		{
+			handle.lock().unwrap().seek(SeekFrom::Start(entry.location))?;
```

Comment:
Shouldn't we handle this gracefully?

Reply:
That is a locking operation rather than error handling, we are locking the handle or waiting for the lock to be free before we read from the handle

Reply:
I get that, but this means the code panics when the mutex is poisened

Reply:
Oh yeah, I will add a new error type for poisoned mutexes, an `InternalError::SyncError`